r/R/dplyr.R

```diff
@@ -410,6 +410,17 @@ build_function_list <- function(FUN) {
         options = list(null_encoding_behavior = null_encoding_behavior)
       )
     },
+    strsplit = function(x, split, fixed = FALSE, perl = FALSE, useBytes = FALSE){
+      
+      regex_metachars <- c(".", "\\", "|", "(", ")", "[", "{", "^", "$", "*", "+", "?")
+      is_regex <- any(map_lgl(regex_metachars, ~grepl(.x, split, fixed = TRUE)))
```

Comment:
I think you can simplify this like so
```suggestion
      is_regex <- grepl("[.\\|()[{^$*+?]", split)
```
this should work because in a regular expression, inside a square bracket character class, these special characters are interpreted as literals (as explained at https://stackoverflow.com/a/19976308/375432)

But please sanity check this; I could be missing something

Reply:
Above here, I think it would be worthwhile to assert that `split` is a length 1 character vector, because in R I think `strsplit()` allows `split` to be a vector with length > 1

Reply:
That seems to work and will do!